rust/src/scannerctl/main.rs

```diff
@@ -84,63 +105,32 @@ async fn main() {
     }
 }
 
-async fn run(matches: &ArgMatches) -> Result<(), CliError> {
-    if let Some(result) = feed::run(matches).await {
-        return result;
-    }
-    if let Some(result) = syntax::run(matches).await {
-        return result;
-    }
-    if let Some(result) = execute::run(matches).await {
-        return result;
-    }
-    if let Some(result) = scanconfig::run(matches).await {
-        return result;
+async fn run(action: Action) -> Result<(), CliError> {
+    match action {
+        Action::Syntax(args) => syntax::run(args).await,
+        Action::ScanConfig(args) => scan_config::run(args).await,
+        Action::Osp(args) => osp::run(args).await,
+        Action::Execute(args) => execute::run(args).await,
+        Action::NotusUpdate(args) => notus_update::scanner::run(args).await,
+        Action::Feed(args) => feed::run(args).await,
     }
-    if let Some(result) = notusupdate::scanner::run(matches).await {
-        return result;
-    }
-    if let Some(result) = osp::run(matches).await {
-        return result;
-    }
-    Err(CliError {
-        filename: "".to_string(),
-        kind: CliErrorKind::Corrupt(format!(
-            "No valid subcommand found: {:?}",
-            matches.subcommand()
-        )),
-    })
 }
 
-pub fn set_logging(level: u8) {
-    let lv = if level > 1 {
-        tracing::Level::TRACE
-    } else if level > 0 {
-        tracing::Level::DEBUG
-    } else {
-        tracing::Level::INFO
+fn set_logging(level: u8, quiet: bool) {
+    let level = match level {
+        level if level > 1 => Level::TRACE,
+        1 => Level::DEBUG,
+        0 => Level::INFO,
+        _ => {
+            if quiet {
+                Level::ERROR
+            } else {
+                Level::INFO
+            }
+        }
     };
     tracing_subscriber::fmt()
         .with_writer(std::io::stderr)
```

Comment:
This is incorrect, only errors and warnings should be in stderr while info and debug should be in stdout.

The reason for that is that scannerctl can be used in a CI or script like setting in which you want to capture information to process further.